tools/sense_studio/static/testing.js

```diff
@@ -36,18 +36,32 @@ async function startTesting(url) {
     let classifier = document.getElementById('classifier').value;
     let webcamInput = document.getElementsByName('inputSource')[0];
     let saveVideo = document.getElementById('saveVideo');
-    let inputVideoPath = (webcamInput.checked) ? '' : document.getElementById('inputVideoPath').value;
-    let outputVideoName = (saveVideo.checked) ? document.getElementById('outputVideoName').value : '';
+    let inputVideoPath = document.getElementById('inputVideoPath');
+    let inputVideoPathValue = (webcamInput.checked) ? '' : inputVideoPath.value;
+    let inputVideoPathLabel = document.getElementById('inputVideoPathLabel');
+    let outputVideoName = document.getElementById('outputVideoName');
+    let outputVideoNameValue = (saveVideo.checked) ? outputVideoName.value : '';
+    let outputVideoNameLabel = document.getElementById('outputVideoNameLabel');
     let path = document.getElementById('path').value;
     let title = document.getElementById('title').value;
     let buttonTest = document.getElementById('btnTest');
     let buttonCancelTest = document.getElementById('btnCancelTest');
     let frame = document.getElementById('frame');
 
+    if (!inputVideoPathValue || !inputVideoPathValue.endsWith('.mp4')) {
+        setFormWarning(inputVideoPathLabel, inputVideoPath, 'Please provide a valid .mp4 file');
+        return false;
+    }
```

Comment:
This part currently doesn't allow to stream from the webcam, because in that case the `inputVideoPathValue` is empty

Reply:
> Also on the other pages we usually check those conditions already on input and disable/enable the action button accordingly. Would be nice to keep the behavior similar here.

This is too complicated as both the event `input_video` and `save_output_video` are independent of each other so if they are being manipulated individually the solution of doing everything in `oninput` is good. But, if they are both selected together then the functionality created for both individually will make it complicated.

I tried many ways to solve this but could not if we want have it in `oninput` event listeners.

In simple terms, two processes racing for a resource in parallel but there is no lock system.

Reply:
I am processing all conditions at the same time now, which should solve the problem